backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt

```diff
@@ -1251,6 +1251,7 @@ internal class FunctionGenerationContext(val function: LLVMValueRef,
                 returnType == voidType -> {
                     releaseVars()
                     assert(returnSlot == null)
+                    call(context.llvm.Kotlin_mm_safePointFunctionEpilogue, emptyList())
```

Comment:
While these calls are probably inlined in release mode, I'm not sure this happens in debug mode.
If it isn't, we probably should emit the calls only for the new MM.

Reply:
They are marked as `always_inline`, so they should.

Reply:
> I'm not sure this happens in debug mode.

Huh, you're right. I thought `always_inline` would definitely help, but there are many instances (mostly in stdlib) where calls to `Kotlin_mm_safePointFunctionEpilogue` get left in place.

Reply:
That's strange. Is there any other `always_inline` functions that are not inlined?

Reply:
Oh, I see. We link object files in debug mode, thus these callees cannot be inlined.

Reply:
> Oh, I see. We link object files instead of bitcode in debug mode, thus these callees cannot be inlined.

Do we also link object files when compiling via konanc cli?

Reply:
Not by default. Only if we pass `-g -Xcache-directory=path_to/macos_x64g-STATIC`.

Reply:
Then we have 2 problems:
* `always_inline` may not inline even when compiling bitcode. And the [docs](https://clang.llvm.org/docs/AttributeReference.html#always-inline-force-inline) actually don't promise anything
* We may not inline with caches (i.e. actual user side case), because that requires LTO, which will slow down compilation.